2022Swerve/src/main/java/frc/robot/Constants.java

```diff
@@ -12,4 +16,54 @@
  * <p>It is advised to statically import this class (or one of its inner classes) wherever the
  * constants are needed, to reduce verbosity.
  */
-public final class Constants {}
+public final class Constants {
+
+    public static final class OIConstants{
+        //Constants for Operator Interface
+        public static final double kDeadband = 0;
+
+        public static final int kDriverControllerPort = 0;
+
+        public static final int kDriverYAxis = 0;
+        public static final int kDriverXAxis = 0;
+        public static final int kDriverRotAxis = 0;
+
+        public static final int kDriverFODButtonID = 0; //Field Oriented Drive toggle button
+        public static final int kDriveZeroButtonID = 0;
+        
+    }
+
+    public static final class MotorDefaults{
+        //Constants to use as default values for Motor Controllers
+        public static final int kCurrentLimit = 40;
+        public static final double kOpenLoopRampRate = 0.2;
+    }
+
+    public static final class DriveConstants{ //TODO: Update these constants with physical values
+        public static final double kPhysicalMaxSpeed = 0; //Max drivebase speed in meters per second
+        public static final double kPhysicalMaxAngularSpeed = 0; //Max drivebase angular speed in radians per second
+        public static final double kMaxAcceleration = 0; //Max allowed acceleration in units per second (Not a physical value)
+        public static final double kMaxAngluarAcceleration = 0; //Max allowed angluar acceleration in units per second (Not a physical value)
+        public static final double kSpeedFactor = 4;  //Factor to divide the physical max speed by to use as max speed in mapping
+        public static final double kAngularSpeedFactor = 4; //Factor to divide the physical max speed by to use as max speed in mapping
+
+        public static final double kTrackWidth = Units.inchesToMeters(0); //Distance between right and left wheels
+        public static final double kWheelBase = Units.inchesToMeters(0); //Distance between front and back wheels
+        public static final SwerveDriveKinematics kDriveKinematics = new SwerveDriveKinematics( //Creates robot geometry using the locations of the 4 wheels
+            new Translation2d(kWheelBase / 2, -kTrackWidth / 2), 
+            new Translation2d(kWheelBase / 2, kTrackWidth /2),
+            new Translation2d(-kWheelBase / 2, -kTrackWidth / 2),
+            new Translation2d(-kWheelBase / 2, kTrackWidth / 2));
+    }
+
+    public static final class SwerveModuleConstants{ //TODO: Update these constants with physical values
+        public static final double kWheelDiameterMeters = Units.inchesToMeters(4);
+        public static final double kDriveMotorGearRatio = 1 / 1;
+        public static final double kTurningMotorGearRatio = 1 / 1;
```

Comment:
We know the gear ratios:

https://austincans2158.slack.com/archives/CCVA38MH8/p1661381841554289

6.75:1